lib/auth/keystore/gcp_kms.go

```diff
@@ -228,65 +228,96 @@ func (g *gcpKMSKeyStore) canSignWithKey(ctx context.Context, raw []byte, keyType
 	return true, nil
 }
 
-// DeleteUnusedKeys deletes all keys from KMS if they are:
-// 1. Labeled by this server (matching HostUUID) when they were created
-// 2. Not included in the argument activeKeys
+// DeleteUnusedKeys deletes all keys from the configured KMS keyring if they:
+//  1. Are not included in the argument activeKeys
+//  2. Are labeled with hostLabel (teleport_auth_host)
+//  3. The hostLabel value matches the local host UUID
+//
+// The activeKeys argument is meant to contain to complete set of raw key IDs as
+// stored in the current CA specs in the backend.
+//
+// The reason this does not delete any keys created by a different auth server
+// is to avoid a case where:
+// 1. A different auth server (auth2) creates a new key in GCP KMS
+// 2. This function (running on auth1) deletes that new key
+// 3. auth2 saves the id of this deleted key to the backend CA
+//
+// or a simpler case where: the other auth server is running in a completely
+// different Teleport cluster and the keys it's actively using will never appear
+// in the activeKeys argument
 func (g *gcpKMSKeyStore) DeleteUnusedKeys(ctx context.Context, activeKeys [][]byte) error {
 	// Make a map of currently active key versions, this is used for lookups to
-	// check which keys in KMS are unused, and holds a count of how many times
-	// they are found in KMS. If any active keys are not found in KMS, we are in
-	// a bad state, so key deletion will be aborted.
+	// check which keys in KMS are unused.
 	activeKmsKeyVersions := make(map[string]int)
 	for _, activeKey := range activeKeys {
-		keyID, err := parseGCPKMSKeyID(activeKey)
+		keyIsRelevant, err := g.canSignWithKey(ctx, activeKey, keyType(activeKey))
 		if err != nil {
-			// could be a different type of key
+			return trace.Wrap(err)
+		}
+		if !keyIsRelevant {
+			// Ignore active keys that are not GCP KMS keys or are in a
+			// different keyring than the one this Auth is configured to use.
 			continue
 		}
+		keyID, err := parseGCPKMSKeyID(activeKey)
+		if err != nil {
+			// Realistically we should not hit this since canSignWithKey already
+			// calls parseGCPKMSKeyID.
+			return trace.Wrap(err)
+		}
 		activeKmsKeyVersions[keyID.keyVersionName] = 0
 	}
 
-	var unusedKeyIDs []gcpKMSKeyID
+	var keysToDelete []gcpKMSKeyID
 
 	listKeyRequest := &kmspb.ListCryptoKeysRequest{
 		Parent: g.keyRing,
-		Filter: fmt.Sprintf("labels.%s=%s", hostLabel, g.hostUUID),
+		// Only bother listing keys created by Teleport which should have the
+		// hostLabel set
+		Filter: fmt.Sprintf("labels.%s:*", hostLabel),
```

Comment:
nit. 
```suggestion
		// Only bother listing keys created by Teleport which should have the
		// hostLabel set.
		Filter: fmt.Sprintf("labels.%s:*", hostLabel),
```
Also assuming `:*` checks if a key is defined?